dias/analyzers/flag_rfi_analyzer.py

```diff
@@ -0,0 +1,592 @@
+
+                    within_range = np.flatnonzero((rdr.freq >= self.freq_low) &
+                                                  (rdr.freq <= self.freq_high))
+                    freq_sel = slice(within_range[0], within_range[-1]+1)
+
+                    stack_sel = [ii for ii, pp in
+                                 enumerate(rdr.prod[rdr.stack['prod']])
+                                 if pp[0] == pp[1]]
+
+                    # Load autocorrelations
+                    t0 = time.time()
+
+                    data = andata.CorrData.from_acq_h5(
+                                        files,
+                                        datasets=['vis', 'flags/vis_weight'],
+                                        freq_sel=freq_sel, stack_sel=stack_sel,
+                                        apply_gain=False, renormalize=False)
+
+                    tspan = (time.time() - t0,)
+                    self.logger.info("Took %0.1f seconds " % tspan +
+                                     "to load autocorrelations.")
+                    t0 = time.time()
+
+                    # Construct RFI mask for each cylinder/polarisation
+                    cyl_index, cyl_auto, cyl_ndev = rfi.number_deviations(
+                                    data,
+                                    stack=False,
+                                    apply_static_mask=self.apply_static_mask,
+                                    freq_width=self.freq_width,
+                                    time_width=self.time_width,
+                                    rolling=self.rolling)
+
+                    # Construct RFI mask for stacked incoherent beam
+                    index, auto, ndev = rfi.number_deviations(
+                                    data,
+                                    stack=True,
+                                    apply_static_mask=self.apply_static_mask,
+                                    freq_width=self.freq_width,
+                                    time_width=self.time_width,
+                                    rolling=self.rolling)
+
+                    tspan = (time.time() - t0,)
+                    self.logger.info("Took %0.1f seconds " % tspan +
+                                     "to generate mask.")
+
+                    # Concatenate and define stack axis
+                    auto = np.concatenate((cyl_auto, auto), axis=1)
+                    ndev = np.concatenate((cyl_ndev, ndev), axis=1)
+
+                    mask_missing = auto > 0.0
+                    mask_before = np.isfinite(ndev)
+                    mask_after = ndev <= self.threshold_mad
+
+                    stack = [POL_MAP[inputmap[ii].pol] + '-' +
+                             CYL_MAP[inputmap[ii].cyl] for ii in cyl_index]
+                    stack.append('ALL')
+
+                    # Determine name of ouput file
+                    epos = acq.name.find(self.instrument)+len(self.instrument)
```

Comment:
These three lines, the try/catch block for `makedirs`  and `acq_start` could be outside the two inner loops.